transforms/helpers/options.ts

```diff
@@ -1,17 +1,71 @@
+import type { ZodError } from 'zod';
+import { z } from 'zod';
+import logger from './log-helper';
 import type { RuntimeData } from './runtime-data';
 
-export interface UserOptions {
+export const UserOptionsSchema = z.object({
```

Comment:
👍 nice

Reply:
I've tried several of these shape validation libraries and `zod` feels the nicest with TS by far. If you know TS, the APIs are super logical. I barely had to look at the docs at all once I figured out the patterns.